airflow/secrets/metastore.py

```diff
@@ -31,12 +31,25 @@ class MetastoreBackend(BaseSecretsBackend):
 
     # pylint: disable=missing-docstring
     @provide_session
-    def get_connections(self, conn_id, session=None) -> List['Connection']:
+    def get_connection(self, conn_id, session=None) -> Optional['Connection']:
         from airflow.models.connection import Connection
 
-        conn_list = session.query(Connection).filter(Connection.conn_id == conn_id).all()
+        conn = session.query(Connection).filter(Connection.conn_id == conn_id).first()
         session.expunge_all()
-        return conn_list
+        return conn
+
+    # pylint: disable=missing-docstring
+    @provide_session
+    def get_connections(self, conn_id, session=None) -> List['Connection']:
+        warnings.warn(
+            "This method is deprecated. Please use "
+            "`airflow.secrets.metastore.MetastoreBackend.get_connection`.",
+            PendingDeprecationWarning,
```

Comment:
```suggestion
            PendingDeprecationWarning,
            stacklevel=3,
```

(+1 here because of the decorator.)